internal/charmstore/migrations_test.go

```diff
@@ -822,49 +1471,104 @@ func (s *migrationsSuite) checkBaseEntitiesCount(c *gc.C, expectCount int) {
 	c.Assert(count, gc.Equals, expectCount)
 }
 
-func (s *migrationsSuite) insertEntity(c *gc.C, id *charm.Reference, name string, size int64) {
-	entity := &mongodoc.Entity{
-		URL:     id,
-		BaseURL: baseURL(id),
-		Name:    name,
-		Size:    size,
+// insertEntity creates a new entity in the database based on the
+// provided mongodoc.Entity. As a convenience some fields will
+// automatically derived if they are set to their zero value, these are:
+//
+//     BaseURL
+//     Name
+//     Revision
+//     Series
+//     User
+//     PromulgatedRevision
+//
+// After insertion any name in fields will be removed from the inserted
+// document.
+func (s *migrationsSuite) insertEntity(c *gc.C, entity *mongodoc.Entity, fields ...string) {
+	if entity.BaseURL == nil {
```

Comment:
We could probably just set these values unconditionally. Is there anywhere we want to knowingly insert an internally inconsistent entity?